src/librustc_codegen_ssa/back/symbol_export.rs

```diff
@@ -178,7 +178,7 @@ fn exported_symbols_provider_local(
         .map(|(&def_id, &level)| (ExportedSymbol::NonGeneric(def_id), level))
         .collect();
 
-    if tcx.entry_fn(LOCAL_CRATE).is_some() {
+    if tcx.entry_fn(()).is_some() {
```

Comment:
Can we give a name (`LocalCrate`) to the unit struct that implements `query::Key`? Otherwise, it's not immediately clear to me what this line is doing.

Reply:
That seems like a good idea to me. @eddyb do you agree?

Reply:
That's one way. But this is basically a nulary query.
If we tweaked the macros a bit we could write this instead:
```rust
if tcx.entry_fn().is_some() {
```

Reply:
If we go that route, we should rename the `entry_fn` query to `entry_fn_local_crate`. Otherwise the same readability issues apply.

Reply:
I wonder if adding an improved doc comment would suffice? It seems reasonable clear to me that the `entry_fn` computes, well, the entry function -- I guess the thing that is sort of non-obvious is the overall model where the root crate has an entry function, and it seems like that would be best explained in docs? 

In particular, `entry_fn_for_local_crate` (if we're going to use a long name, I'd prefer to make it read well...) seems to imply that other crates can have entry functions, which isn't really true?

Anyway, a minor thing, but I guess it may become a more wide-spread convention either way.